flytekit/models/core/workflow.py

```diff
@@ -778,13 +801,19 @@ def on_failure(self):
         """
         return self._on_failure
 
+    @property
+    def tags(self):
+        """
+        :rtype: flytekit.models.core.workflow.WorkflowMetadata.Tags
+        """
+        return self._tags
+
     def to_flyte_idl(self):
         """
         :rtype: flyteidl.core.workflow_pb2.WorkflowMetadata
         """
-        workflow_metadata = _core_workflow.WorkflowMetadata()
-        if self.on_failure:
-            workflow_metadata.on_failure = self.on_failure
+
+        workflow_metadata = flyteidl.core.WorkflowMetadata(self.on_failure or 0, self.tags or {})
```

Comment:
Is setting `on_failure` default value to `flyteidl.workflow_metadata.OnFailurePolicy.FailImmediately` ok?